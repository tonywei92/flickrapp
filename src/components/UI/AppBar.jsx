import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { fade, makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import InputBase from '@material-ui/core/InputBase';
import IconButton from '@material-ui/core/IconButton';
import InputAdornment from '@material-ui/core/InputAdornment';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing(1),
      width: 'auto',
    },
  },
  searchIcon: {
    color: 'inherit',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 0),
    // vertical padding + font size from searchIcon
    paddingLeft: '20px',
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: '12ch',
      '&:focus': {
        width: '20ch',
      },
    },
  },
}));

const Navbar = (props) => {
  const {
    title,
    buttons,
    searchInputValue,
    onSearchInputChanged,
    onSearchSubmit,
    onMenuClick,
  } = props;
  const classes = useStyles();

  const [searchOpen, setSearchOpen] = useState(!!searchInputValue);
  const openSearch = (open) => {
    if (!open) {
      if (!searchInputValue.trim()) {
        setSearchOpen(open);
      }
    } else {
      setSearchOpen(open);
    }
  };

  const handleSearchKeyUp = (e) => {
    if (e.keyCode === 13) {
      onSearchSubmit();
    }
    if (e.keyCode === 27) {
      openSearch(false);
    }
  };

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="menu"
            onClick={onMenuClick}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" className={classes.title}>
            {!searchOpen && title}
          </Typography>
          {searchOpen && (
            <div className={classes.search}>
              <InputBase
                autoFocus
                onKeyUp={handleSearchKeyUp}
                onBlur={() => {
                  openSearch(false);
                }}
                placeholder="Searchâ€¦"
                classes={{
                  root: classes.inputRoot,
                  input: classes.inputInput,
                }}
                value={searchInputValue}
                onChange={onSearchInputChanged}
                inputProps={{ 'aria-label': 'search' }}
                endAdornment={
                  <InputAdornment position="end">
                    <IconButton
                      aria-label="toggle password visibility"
                      className={classes.searchIcon}
                      onClick={onSearchSubmit}
                    >
                      <SearchIcon />
                    </IconButton>
                  </InputAdornment>
                }
              />
            </div>
          )}

          {!searchOpen && (
            <IconButton
              aria-label="toggle password visibility"
              className={classes.searchIcon}
              color="inherit"
              onClick={() => openSearch(true)}
            >
              <SearchIcon />
            </IconButton>
          )}
          {buttons}
        </Toolbar>
      </AppBar>
    </div>
  );
};

Navbar.defaultProps = {
  buttons: [],
  searchInputValue: '',
  onSearchInputChanged: () => {},
  onSearchSubmit: () => {},
  onMenuClick: () => {},
};

Navbar.propTypes = {
  title: PropTypes.string.isRequired,
  buttons: PropTypes.arrayOf(PropTypes.node),
  searchInputValue: PropTypes.string,
  onSearchInputChanged: PropTypes.func,
  onSearchSubmit: PropTypes.func,
  onMenuClick: PropTypes.func,
};

export default Navbar;
