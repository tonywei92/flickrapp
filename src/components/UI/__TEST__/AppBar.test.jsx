import React from 'react';
import renderer from 'react-test-renderer';
import AppBar from '../AppBar';

test('Render correct carousel structure', () => {
  const buttons = [
    <button type="button" key="button1">
      button 1
    </button>,
    <button type="button" key="button2">
      button 2
    </button>,
  ];
  let component = renderer.create(
    <AppBar
      title="test 1"
      buttons={buttons}
      searchInputValue="test input value"
      onSearchInputChanged={() => {}}
      onSearchSubmit={() => {}}
      onMenuClick={() => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <AppBar
      title="test 1"
      buttons={buttons}
      onSearchInputChanged={() => {}}
      onSearchSubmit={() => {}}
      onMenuClick={() => {}}
    />
  );

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
