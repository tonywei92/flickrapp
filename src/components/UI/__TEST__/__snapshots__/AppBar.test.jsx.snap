// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correct carousel structure 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      />
      <div
        className="makeStyles-search-4"
      >
        <div
          className="MuiInputBase-root makeStyles-inputRoot-6 MuiInputBase-adornedEnd"
          onClick={[Function]}
        >
          <input
            aria-label="search"
            autoFocus={true}
            className="MuiInputBase-input makeStyles-inputInput-7 MuiInputBase-inputAdornedEnd"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            placeholder="Searchâ€¦"
            type="text"
            value="test input value"
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              aria-label="toggle password visibility"
              className="MuiButtonBase-root MuiIconButton-root makeStyles-searchIcon-5"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <button
        type="button"
      >
        button 1
      </button>
      <button
        type="button"
      >
        button 2
      </button>
    </div>
  </header>
</div>
`;

exports[`Render correct carousel structure 2`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        test 1
      </h6>
      <button
        aria-label="toggle password visibility"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-searchIcon-5 MuiIconButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        type="button"
      >
        button 1
      </button>
      <button
        type="button"
      >
        button 2
      </button>
    </div>
  </header>
</div>
`;
